src/main/java/io/codelex/java/advanced/test/Exercise3/Joiner.java

```diff
@@ -0,0 +1,26 @@
+package io.codelex.java.advanced.test.Exercise3;
+
+public class Joiner <T> {
+
+    @SafeVarargs
+    public final String join(T... args) {
+        if (args.length == 0) {
+            return "";
+        }
+        Class<?> firstElementType = args[0].getClass();
+        for (int i = 1; i < args.length; i++) {
+            if (args[i].getClass() != firstElementType) {
```

Comment:
Šādas pārbaudes nav nepieciešamas - ja reiz Joiner norādīsi generic tipu, piemēram Employee, tad join metodei neko citu padot nevarēs izņemot Employee vai tā child klases un padot child klasi būtu ok, to nevajag ierobežot ar šādu pārbaudi.